pkg/cluster/delete.go

```diff
@@ -286,6 +286,15 @@ func (m *manager) Delete(ctx context.Context) error {
 
 	m.log.Printf("deleting resources")
 	err = m.deleteResources(ctx)
```

Comment:
I think you want to remove this call as if `m.deleteResources(ctx)` is successful, it will attempt to run again no matter what.  

Should only need the `m.deleteResources(ctx)` within the retry loop.

Reply:
left that in accidentally.

closing this PR as https://github.com/Azure/ARO-RP/pull/1483 fixes the same problem